code/modules/mob/living/carbon/xenomorph/abilities/shrieker/shrieker_powers.dm

```diff
@@ -0,0 +1,219 @@
+
+	if (dist > range)
+		to_chat(X, SPAN_WARNING("[H] is too far away!"))
+		return
+
+	if (dist > 1)
+		var/turf/targetTurf = get_step(X, get_dir(X, H))
+		if (targetTurf.density)
+			to_chat(X, SPAN_WARNING("You can't attack through [targetTurf]!"))
+			return
+		else
+			for (var/atom/I in targetTurf)
+				if (I.density && !I.throwpass && !istype(I, /obj/structure/barricade) && !istype(I, /mob/living))
+					to_chat(X, SPAN_WARNING("You can't attack through [I]!"))
+					return
+
+
+	if (H.stat == DEAD)
+		to_chat(X, SPAN_XENODANGER("[H] is dead, why would you want to touch it?"))
+		return
+
+	if (X.mutation_type == SHRIEKER_NORMAL)
+		var/datum/behavior_delegate/shrieker/BD = X.behavior_delegate
+		if (!istype(BD))
+			return
+
+
+
+	// Flick overlay and play sound
+	X.animation_attack_on(A, 10)
+	var/S = pick('sound/weapons/punch1.ogg','sound/weapons/punch2.ogg','sound/weapons/punch3.ogg','sound/weapons/punch4.ogg')
+	playsound(H,S, 50, 1)
+
+	var/root_duration = 1 SECONDS
+	var/damage = 15
+	X.visible_message(SPAN_XENODANGER("[X] extends with its claws and smashes [A], pinning them to the ground dealing damage!"), SPAN_XENOHIGHDANGER("You extend your claws and smash [A], pinning them to the ground and dealing damage!"))
+
+	H.frozen = TRUE
+	H.update_canmove()
+	H.apply_armoured_damage(damage, ARMOR_MELEE, BRUTE, "chest", 15)
+
+	if (ishuman(H))
+		var/mob/living/carbon/human/target = H
+		target.update_xeno_hostile_hud()
+
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/unroot_human, H), get_xeno_stun_duration(H, root_duration))
+		to_chat(H, SPAN_XENOHIGHDANGER("[X] has pinned you to the ground! You cannot move!"))
+
+	apply_cooldown()
+	..()
+	return
+
+/datum/action/xeno_action/activable/cleave/proc/remove_root()
+	buffed = FALSE
+
+/datum/action/xeno_action/activable/pounce/shrieker/additional_effects(mob/living/L)
+
+	var/mob/living/carbon/human/H = L
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(X.mutation_type != SHRIEKER_NORMAL)
+		return
+
+	var/datum/behavior_delegate/shrieker/BD = X.behavior_delegate
+	if (!istype(BD))
+		return
+
+	X.visible_message(SPAN_XENODANGER("The [X] dashes forward, its claws extended!"), SPAN_XENODANGER("You dash forward, extending your claws!"))
+	H.attack_alien(X, 15)
+
+
+/datum/action/xeno_action/activable/acid_throw/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
+
+	if (!istype(X))
+		return
+
+	if (!X.check_state())
+		return
+
+	if (!action_cooldown_check())
+		return
+
+	if(!A || A.layer >= FLY_LAYER || !isturf(X.loc))
+		return
+
+	if (!check_and_use_plasma_owner())
+		return
+
+	if(!check_clear_path_to_target(X, A, TRUE))
+		to_chat(X, SPAN_XENOWARNING("Something is in the way!"))
+		return
+
+	if (X.mutation_type == SHRIEKER_NORMAL)
+		var/datum/behavior_delegate/shrieker/BD = X.behavior_delegate
+		if (!istype(BD))
+			return
+		if (!BD.use_internal_acid_ability(throw_cost))
+			return
+
+	var/turf/T = get_turf(A)
+	var/acid_bolt_message = "a barrage of acid"
```

Comment:
why is this variable a thing? its not changed by anything, just write that out in the string

Reply:
pretty sure its so its configurable. Same thing applies for trapper boiler

Reply:
it isn't configurable, it's in a proc, proc local variables can't be changed in runtime

put it in the ability definition